### PR TITLE
fix(clone): reset index of YADM_WORK

### PR DESCRIPTION
**Pull request by AVM-Martin**
### What does this PR do?

> Re-open PR https://github.com/yadm-dev/yadm/pull/492

Whenever using `clone` command inside YADM_WORK sub-directory, we need
to checkout the correct repo contents.

Steps to reproduce:

```bash
mkdir $HOME/subdir
cd $HOME/subdir

yadm clone --bootstrap "<repo-url>"
yadm status
```

### What issues does this PR fix or reference?

[A list of related issues / pull requests.]

### Previous Behavior

`yadm status` reported a lot of deleted files.

### New Behavior

`yadm status` will report clean working directory.

### Have [tests][1] been written for this change?

Yes

### Have these commits been [signed with GnuPG][2]?

Yes

---

Please review [yadm's Contributing Guide][3] for best practices.

[1]: https://github.com/TheLocehiliosan/yadm/blob/master/.github/CONTRIBUTING.md#test-conventions
[2]: https://help.github.com/en/articles/signing-commits
[3]: https://github.com/TheLocehiliosan/yadm/blob/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: yadm [hunk 1]
```diff
@@ -827,7 +827,7 @@ function clone() {
   rm -rf "$wc"
 
   # then reset the index as the --no-checkout flag makes the index empty
-  "$GIT_PROGRAM" reset --quiet -- .
+  "$GIT_PROGRAM" reset --quiet -- "$YADM_WORK"
```
Comment (erijo) [inline yadm:830]: Consider using `:`/ instead of `$YADM_WORK` to match the pathspec used below (in the checkout command).

**Review by erijo — COMMENTED** [review]
Thanks for you PR. Just a few minor comments that would be good if you could address.

File: test/test_clone.py [hunk 2]
```diff
@@ -326,3 +326,32 @@ def test_no_repo(
...
+
+    # create sub-directory
+    subdir = paths.work.mkdir("subdir")
+
+    # determine remote url
+    remote_url = f"file://{paths.remote}"
+
+    # run the clone command
+    args = ["clone", "-w", paths.work]
+    args += [remote_url]
```
[the remark this answers is not in the document]
Comment (AVM-Martin) [reply test/test_clone.py]: Done in a7939bec7ba89c39bd55471ae33103db9d435c6c

File: test/test_clone.py [hunk 3]
```diff
@@ -326,3 +326,32 @@ def test_no_repo(
...
+    run = runner(command=yadm_cmd(*args), cwd=subdir)
+
+    # clone should succeed, and repo should be configured properly
+    assert successful_clone(run, paths, repo_config)
+
+    # test that the conflicts are preserved in the work tree
```
[the remark this answers is not in the document]
Comment (AVM-Martin) [reply test/test_clone.py]: Fixed in 85e8c1ddfcbb7d206cc6945fc1fdf25e100c9b4a, english is not my first language

File: yadm (hunk 1 above)
Comment (AVM-Martin) [reply yadm]: Nice info, I just know about pathspec
Changed in ae3a1494497702d16f2331a8d39c136c6e11f066

**Comment by erijo** [comment]
Thanks for your contribution @AVM-Martin! :+1:

**Pull request merged**